lib/offsite_payments/integrations/paytm.rb

```diff
@@ -0,0 +1,172 @@
+module OffsitePayments #:nodoc:
+  module Integrations #:nodoc:
+    module Paytm
+    
+      def self.notification(post, options = {})
+        Notification.new(post, options)
+      end
+
+      def self.return(query_string, options = {})
+        Return.new(query_string, options)
+      end
+
+      def self.sign(fields, key)
+        Base64.encode64(OpenSSL::HMAC.digest(OpenSSL::Digest::SHA256.new, key, fields.sort.join)).delete("\n")
+      end
+
+      class Helper < OffsitePayments::Helper
+        CURRENCY_SPECIAL_MINOR_UNITS = {
+          'BIF' => 0,
+          'BYR' => 0,
+          'CLF' => 0,
+          'CLP' => 0,
+          'CVE' => 0,
+          'DJF' => 0,
+          'GNF' => 0,
+          'HUF' => 0,
+          'ISK' => 0,
+          'JPY' => 0,
+          'KMF' => 0,
+          'KRW' => 0,
+          'PYG' => 0,
+          'RWF' => 0,
+          'UGX' => 0,
+          'UYI' => 0,
+          'VND' => 0,
+          'VUV' => 0,
+          'XAF' => 0,
+          'XOF' => 0,
+          'XPF' => 0,
+          'BHD' => 3,
+          'IQD' => 3,
+          'JOD' => 3,
+          'KWD' => 3,
+          'LYD' => 3,
+          'OMR' => 3,
+          'TND' => 3,
+          'COU' => 4
+        }
+
+        def initialize(order, account, options = {})
+
+          @forward_url_stag = 'https://pguat.paytm.com/oltp-web/genericPT?MID='+account.to_s
```

Comment:
The account param here will need to be URI encoded. If not this will allow anyone to add arbitrary parameters to this URL. Example:

```ruby
mid_param = { MID: account }.to_query
@forward_url_stag = "https://pguat.paytm.com/oltp-web/genericPT?#{mid_param}"
```